src/main/java/teammates/ui/webapi/UpdateNotificationAction.java

```diff
@@ -18,23 +19,20 @@ public class UpdateNotificationAction extends AdminOnlyAction {
 
     @Override
     public JsonResult execute() throws InvalidHttpRequestBodyException {
-        String notificationId = getNonNullRequestParamValue(Const.ParamsNames.NOTIFICATION_ID);
+        UUID notificationId = UUID.fromString(getNonNullRequestParamValue(Const.ParamsNames.NOTIFICATION_ID));
         NotificationUpdateRequest notificationRequest = getAndValidateRequestBody(NotificationUpdateRequest.class);
 
         Instant startTime = Instant.ofEpochMilli(notificationRequest.getStartTimestamp());
         Instant endTime = Instant.ofEpochMilli(notificationRequest.getEndTimestamp());
 
-        UpdateOptions newNotification = NotificationAttributes.updateOptionsBuilder(notificationId)
-                .withStartTime(startTime)
-                .withEndTime(endTime)
-                .withStyle(notificationRequest.getStyle())
-                .withTargetUser(notificationRequest.getTargetUser())
-                .withTitle(notificationRequest.getTitle())
-                .withMessage(notificationRequest.getMessage())
-                .build();
-
         try {
-            return new JsonResult(new NotificationData(logic.updateNotification(newNotification)));
+            Notification updateNotification = sqlLogic.updateNotification(notificationId, startTime, endTime,
+                    notificationRequest.getStyle(), notificationRequest.getTargetUser(), notificationRequest.getTitle(),
+                    notificationRequest.getMessage());
+
+            HibernateUtil.flushSession();
```

Comment:
No need to flush (unless we need a generated (in db) value such as ID, createdAt/updatedAt timestamp)